src/platform/silabs/rs911x/BLEManagerImpl.cpp

```diff
@@ -120,39 +121,38 @@ void BLEManagerImpl::ProcessEvent(SilabsBleWrapper::BleEvent_t inEvent)
     switch (inEvent.eventType)
     {
     case SilabsBleWrapper::BleEventType::RSI_BLE_CONN_EVENT: {
-        BLEMgrImpl().HandleConnectEvent((inEvent.eventData));
+        BLEMgrImpl().HandleConnectEvent(&inEvent.eventData);
         // Requests the connection parameters change with the remote device
-        rsi_ble_conn_params_update(inEvent.eventData->resp_enh_conn.dev_addr, BLE_MIN_CONNECTION_INTERVAL_MS,
+        rsi_ble_conn_params_update(inEvent.eventData.resp_enh_conn.dev_addr, BLE_MIN_CONNECTION_INTERVAL_MS,
                                    BLE_MAX_CONNECTION_INTERVAL_MS, BLE_SLAVE_LATENCY_MS, BLE_TIMEOUT_MS);
-        rsi_ble_set_data_len(inEvent.eventData->resp_enh_conn.dev_addr, RSI_BLE_TX_OCTETS, RSI_BLE_TX_TIME);
+        rsi_ble_set_data_len(inEvent.eventData.resp_enh_conn.dev_addr, RSI_BLE_TX_OCTETS, RSI_BLE_TX_TIME);
 
         // Used to send the Indication confirmation
-        memcpy(dev_address, inEvent.eventData->resp_enh_conn.dev_addr, RSI_DEV_ADDR_LEN);
-        ble_measurement_hndl = inEvent.eventData->rsi_ble_measurement_hndl;
+        memcpy(dev_address, inEvent.eventData.resp_enh_conn.dev_addr, RSI_DEV_ADDR_LEN);
     }
     break;
     case SilabsBleWrapper::BleEventType::RSI_BLE_DISCONN_EVENT: {
         // event invokes when disconnection was completed
-        BLEMgrImpl().HandleConnectionCloseEvent(inEvent.eventData);
+        BLEMgrImpl().HandleConnectionCloseEvent(&inEvent.eventData);
```

Comment:
Instead of passing a pointer each time, can we modify the function signature to use a `const &`? This avoid ptr validation and reduces the increase of hardfaults slipping in.

This applies to all the `BLEMgrImpl` functions

Reply:
Done